File: tests/rolling_value_at_risk_test.py
import numpy as np
from taflow import RollingValueAtRisk
def test_lifecycle():
    state=RollingValueAtRisk(np.array([],dtype=float),3); state.extend([1.0,-1.0,0.0]); assert state.value is not None; state.reset(); assert len(state)==0

File: crates/taflow-core/src/indicators/rolling_value_at_risk_test.rs
use super::rolling_value_at_risk::RollingValueAtRisk;

#[test]
fn scalar_replay_and_reset_are_invariant() {
    let values = [1.0, -1.0, 2.0];
    let mut state = RollingValueAtRisk::new(3, 0.95).unwrap();

    assert_eq!(state.append(values[0]), None);
    assert_eq!(state.append(values[1]), None);
    let expected = state.append(values[2]);
    assert!(expected.is_some());
    assert_eq!(state.value(), expected);

    state.reset();
    assert_eq!(state.value(), None);
    assert_eq!(state.append(values[0]), None);
    assert_eq!(state.append(values[1]), None);
    assert_eq!(state.append(values[2]), expected);
}
